src/include/uapi/ipc/dai-intel.h

```diff
@@ -78,6 +78,7 @@
 
 /* SSP Configuration Request - SOF_IPC_DAI_SSP_CONFIG */
 struct sof_ipc_dai_ssp_params {
+	struct sof_ipc_hdr hdr;
```

Comment:
the ssp/dmic/hda_params should not include the header, they are subparts of dai_config which already has a header.

Reply:
The intention here is that all sub structures (in this case params) has a size header to help with data parsing/verification. e.g. if we ever need to send an array of params in the future. This "lite" header is only a uint32_t for size. I've added this size header to all sub structures so that any container growth (via ABI MINOR/PATCH changes) can be correctly validated.

This header wont be used for ssp/dmic/hda params atm since params is only used as the tail strusture in dai_params (and no where else), but it's added to cover all bases for the above future proof reasons.